Makefile

```diff
@@ -44,11 +44,13 @@ $(GENERATORBIN):
 $(GENERATORBIN)/generator: $(G_SRCS) | $(GENERATORBIN)
 	@crystal build $(GENERATORDIR)/generator.$(FILEEXT) -o generator/bin/generator
 
-build-generator: $(GENERATORBIN)/generator
+build-generator:
+	@echo "building binary for generator: $(GENERATOR)"
+	@cd $(GENERATORDIR) && (shards check || shards install) && crystal build generator.cr -o bin/generator
 
-generate-exercise: $(GENERATORBIN)/generator
+generate-exercise:
 	@echo "generating spec file for generator: $(GENERATOR)"
-	@generator/bin/generator $(GENERATOR)
+	@cd $(GENERATORDIR) && (shards check || shards install) && crystal generator.cr $(GENERATOR)
```

Comment:
Is it better to build the generator binary, then call it?

Crystal really provides both options - building first, or executing inline.

Reply:
What is the downside of executing inline? As generators are only rarely being called, performance should probably not be an argument (not sure if it is).

Reply:
I'd either:

1. Run it in an interpreted mode
2. Give it a make rule (so you at least get "dirty bit" rebuilds)

But I think running interpreted is fine.

Reply:
@hiljusti - Do you think it is too early to run in interpreted mode?

> The interpreter is currently on an experimental phase, with lots of [missing bits](https://github.com/crystal-lang/crystal/issues/11555). The reason to merge it at this early stage is to enable a proper discussion of interpreter-related PRs and speed up its development a bit. For those willing to try it out, we welcome interpreter-related issues taking into consideration the already known issues aforelinked.

Source: https://crystal-lang.org/2021/12/29/crystal-i.html

Reply:
Maybe, but idk, I'd wait until seeing a problem before changing it. It seems to be working fine so far, and it is just running a single file so if it's working now it might just keep working forever? (i.e. it's not some crazy huge thing with complex dependencies that would stress the edges of their interpreter)